src/shared/components/copyDownloadControls/CopyDownloadControls.tsx

```diff
@@ -132,18 +132,22 @@ export class CopyDownloadControls extends React.Component<ICopyDownloadControlsP
                     className={`${copyDownloadStyles["centered-modal-dialog"]}`}
                 >
                     <Modal.Header>
-                        Download Complete!
+                        {this.showErrorMessage ? "Download Error!" : "Download Complete!"}
                     </Modal.Header>
                     <Modal.Body>
-                        Please click on Copy to copy downloaded data to clipboard.
+                        {this.showErrorMessage && "An error occurred while downloading the data. "}
+                        Please click on Copy to copy the data to clipboard.
                     </Modal.Body>
                     <Modal.Footer>
                         <span
                             ref={(el: HTMLElement|null) => {this._modalCopyButtonContainer = el;}}
                         >
                             <button
                                 ref={(el:HTMLButtonElement) => {this._modalCopyButton = el;}}
-                                onClick={() => {this.copyingData = false;}}
+                                onClick={() => {
+                                    this.copyingData = false;
```

Comment:
you should wrap these in mobx action or put them in a method call with @action decorator.

Reply:
its better in general for these not to be arrow fns because that makes shouldComponentUpdate return true always

Reply:
added an `@action` method and updated `onClick` and `onHide` properties to refer to it:
```
@action
private handleModalClose()
```